Breeze.UI/src/app/shared/services/api.service.ts

```diff
@@ -201,127 +172,122 @@ export class ApiService {
     params.set('feeType', data.feeType);
     params.set('allowUnconfirmed', 'true');
-
     return this.http
       .get(`${this.currentApiUrl}/wallet/maxbalance`, new RequestOptions({ headers: this.headers, search: params }))
       .map((response: Response) => response);
   }
-
   /**
    * Get a wallets transaction history info from the API.
    */
   getWalletHistory(data: WalletInfo): Observable<any> {
     this.getCurrentCoin();
-
     const params: URLSearchParams = new URLSearchParams();
     params.set('walletName', data.walletName);
-
     return Observable
       .interval(this.pollingInterval)
       .startWith(0)
       .switchMap(() => this.http.get(`${this.currentApiUrl}/wallet/history`, new RequestOptions({ headers: this.headers, search: params })))
       .map((response: Response) => response);
   }
-
   /**
    * Get unused receive addresses for a certain wallet from the API.
    */
   getUnusedReceiveAddress(data: WalletInfo): Observable<any> {
     this.getCurrentCoin();
-
     const params: URLSearchParams = new URLSearchParams();
     params.set('walletName', data.walletName);
     params.set('accountName', 'account 0'); // temporary
-
     return this.http
       .get(`${this.currentApiUrl}/wallet/unusedaddress`, new RequestOptions({ headers: this.headers, search: params }))
       .map((response: Response) => response);
   }
-
   /**
    * Estimate the fee of a transaction
    */
   estimateFee(data: FeeEstimation): Observable<any> {
     this.getCurrentCoin();
-
     const params: URLSearchParams = new URLSearchParams();
     params.set('walletName', data.walletName);
     params.set('accountName', data.accountName);
     params.set('destinationAddress', data.destinationAddress);
     params.set('amount', data.amount);
     params.set('feeType', data.feeType);
     params.set('allowUnconfirmed', 'true');
-
     return this.http
       .get(`${this.currentApiUrl}/wallet/estimate-txfee`, new RequestOptions({ headers: this.headers, search: params }))
       .map((response: Response) => response);
   }
-
   /**
    * Build a transaction
    */
   buildTransaction(data: TransactionBuilding): Observable<any> {
     this.getCurrentCoin();
-
     return this.http
       .post(`${this.currentApiUrl}/wallet/build-transaction`, JSON.stringify(data), { headers: this.headers })
       .map((response: Response) => response);
   }
-
   /**
    * Send transaction
    */
   sendTransaction(data: TransactionSending): Observable<any> {
     this.getCurrentCoin();
-
     return this.http
       .post(`${this.currentApiUrl}/wallet/send-transaction`, JSON.stringify(data), { headers: this.headers })
       .map((response: Response) => response);
   }
-
   /**
    * Send shutdown signal to the daemon
    */
   shutdownNode(): Observable<any> {
     this.getCurrentCoin();
-
     return this.http
       .post(`${this.currentApiUrl}/node/shutdown`, '')
       .map((response: Response) => response);
   }
-}
+
+  private onRetryWhen(errors) {
```

Comment:
The `onRetryWhen `method is used in two places. Should be moved into common class